File: 灵茶/图论/Floyd/1334.py
class Solution:
    def findTheCity(self, n: int, edges: List[List[int]], distanceThreshold: int) -> int:
        w = [[inf] * n for _ in range(n)]
        for i ,j, weight in edges:
            w[i][j] = w[j][i] = weight
        
        f = w
        for k in range(n):
            for i in range(n):
                for j in range(n):
                    s = f[i][k] + f[k][j]
                    if s < f[i][j]:
                        f[i][j] = s
        
        ans = 0
        min_cnt = inf
        for i in range(n):
            cnt = 0
            for j in range(n):
                if j != i and f[i][j] <= distanceThreshold:
                    cnt += 1
            if cnt <= min_cnt:
                min_cnt = cnt
                ans = i
        
        return ans

File: 灵茶/图论/Floyd/1334.java
class Solution {
    public int findTheCity(int n, int[][] edges, int distanceThreshold) {
        int[][] graph = new int[n][n];
        for (int[] node : graph){
            Arrays.fill(node, Integer.MAX_VALUE / 2);
        }

        for(int [] edge: edges ){
            graph[edge[0]][edge[1]] = edge[2];
            graph[edge[1]][edge[0]] = edge[2];
        }

        int [][] distance = graph;
        for(int k=0;k<n;k++){
            for( int i = 0; i < n; i++){
                for(int j=0;j<n;j++){
                    distance[i][j] = Math.min(distance[i][j], distance[i][k] + distance[k][j]);
                }
            }
        }

        int ans = 0;
        int minCount = Integer.MAX_VALUE;
        for( int i=0;i<n;i++){
            int count = 0;
            for(int j=0;j<n;j++){
                if( i != j && distance[i][j] <= distanceThreshold){
                    count++;
                }
            }
                          if(count <= minCount){
                    minCount = count;
                    ans = i;
                }
        }
        return ans;
    }
}